### PR TITLE
Display tabs on the module page with the generic method

### PR DESCRIPTION
**Pull request by Quetzacoalt91**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.4.x
| Description?  | Remove module page specific feature for displaying tabs, and use the tab table instead. It also displays the tabs `Installed modules` / `Selection` / `Notifications` in that order from now, as it was planned to change the default page.
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | Nope
| Deprecations? | Yep, `navigation.yml` is not used anymore
| Fixed ticket? | http://forge.prestashop.com/browse/FF-171
| How to test?  | Install a fresh shop, or upgrade from 1.7.3.x. The tabs must have switched.

### Breadcrumb
This PR also improves the breadcrumb of the module page:

* 1.7.3.x
capture du 2018-04-12 11-29-41

* This PR
capture du 2018-04-12 11-29-31

### Performances

As we remove a YML file, performances could be improved on all page running under symfony.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8938)

### CODE REVIEW

**Comment by kpodemski** [comment]
I think that you need keep in mind that access to AdminModules can be customized, i think that every new tab related to this page should inherit those

**Comment by Quetzacoalt91** [comment]
Are you sure about this? It seems the permissions you can change for a group is everything but the `AdminModules` controller.
Although I see a good opportunity to replace `AdminModulesSf` by `AdminModules`, I'm going to keep the access granted on `AdminModulesSf`.

Don't hesitate to tell me if I miss something.

**Comment by Quetzacoalt91** [comment]
Let's keep this PR simple, and leave the old tabs untouched. We only add the new modules subtabs here.

File: classes/Link.php [hunk 1]
```diff
@@ -733,6 +733,9 @@ public function getAdminLink($controller, $withToken = true, $sfRouteParams = ar
             default:
                 $routes = array(
                     'AdminModulesSf' => 'admin_module_manage',
+                    'AdminModulesCatalog' => 'admin_module_catalog',
```
Comment (mickaelandrieu) [inline classes/Link.php:736]: <3

File: install-dev/upgrade/php/ps_1740_update_module_tabs.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        'AdminModulesCatalog' => 'en:Selection|fr:Selection|es:Selección|de:Auswahl|it:Selezione',
+        'AdminModulesManage' => 'en:Installed modules|fr:Modules installés|es:Módulos instalados|de:Installierte Module|it:Moduli installati',
+        'AdminModulesNotifications' => 'en:Notifications|fr:Notifications|es:Notificaciones|de:Nachrichten|it:Notifiche'
+    );
+
+    include_once('add_new_tab.php');
+    foreach ($moduleTabsToBeAdded as $className => $translations) {
+        add_new_tab($className, $translations, 0, false, 'AdminModulesSf');
+    }
+}
```
Comment (mickaelandrieu) [inline install-dev/upgrade/php/ps_1740_update_module_tabs.php]: missing end of line

**Comment by mickaelandrieu** [comment]
It's amazing to see how easy it is to improve this stuff: less code and better organization.

Good job!

**Comment by eternoendless** [comment]
Reviewed 5 of 8 files at r1, 4 of 4 files at r2, 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[admin-dev/themes/default/js/bundle/module/module.js, line 658 at r4](https://beta.reviewable.io/reviews/prestashop/prestashop/8938#-L9zRzoS20TmoLEheFpZ:-L9zRzoS20TmoLEheFp_:bgrpaxo) ([raw file](https://github.com/prestashop/prestashop/blob/9aada29086b7351d14ea861cbee5bbbb0fc59f83/admin-dev/themes/default/js/bundle/module/module.js#L658)):*
> ```JavaScript
>   };
>   
>   this.getNotificationsCount = function () {
> ```

Missing Jsdoc for this method

---

*[admin-dev/themes/default/js/bundle/module/module.js, line 662 at r4](https://beta.reviewable.io/reviews/prestashop/prestashop/8938#-L9zQzLZ489o5OKEUWF-:-L9zQzLZ489o5OKEUWF0:b-hxxuve) ([raw file](https://github.com/prestashop/prestashop/blob/9aada29086b7351d14ea861cbee5bbbb0fc59f83/admin-dev/themes/default/js/bundle/module/module.js#L662)):*
> ```JavaScript
>     var urlToCall = this.baseAdminDir+'module/notifications/count' + token;
> 
>     $.getJSON(urlToCall, function(badge) {
> ```

What if the request fails?

---

*[admin-dev/themes/default/js/bundle/module/module.js, line 665 at r4](https://beta.reviewable.io/reviews/prestashop/prestashop/8938#-L9zR8wf33NEUISl8iU5:-L9zR8wf33NEUISl8iU6:b-397xhj) ([raw file](https://github.com/prestashop/prestashop/blob/9aada29086b7351d14ea861cbee5bbbb0fc59f83/admin-dev/themes/default/js/bundle/module/module.js#L665)):*
> ```JavaScript
>         // TODO: This HTML code comes from an already specific template.
>         // To be moved in a template, with generic classes for badges
>         $('a.tab[href="'+badge.target+'"]').append('<div class="notification-container">\
> ```

```js
'a.tab[href="'+badge.target+'"]'
```
Isn't there another way, like using ids?

---

*[install-dev/data/xml/tab.xml, line 183 at r4](https://beta.reviewable.io/reviews/prestashop/prestashop/8938#-L9zRU1b9jZp7F4Cbq0i:-L9zRU1b9jZp7F4Cbq0j:b-jtv8uh) ([raw file](https://github.com/prestashop/prestashop/blob/9aada29086b7351d14ea861cbee5bbbb0fc59f83/install-dev/data/xml/tab.xml#L183)):*
> ```XML
>                     <tab id="Module_Page_Notifications" id_parent="Module_Page" active="1" hide_host_mode="0" icon="">
>                         <class_name>AdminModulesNotifications</class_name>
>                     </tab>
> ```

There's an indentation issue compared to the old lines

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/prestashop/prestashop/8938)*

**Comment by Quetzacoalt91** [comment]
*[admin-dev/themes/default/js/bundle/module/module.js, line 658 at r4](https://beta.reviewable.io/reviews/prestashop/prestashop/8938#-L9zRzoS20TmoLEheFpZ:-L9zhHFj88HiebTkIcCP:bo1sj8y) ([raw file](https://github.com/prestashop/prestashop/blob/9aada29086b7351d14ea861cbee5bbbb0fc59f83/admin-dev/themes/default/js/bundle/module/module.js#L658)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

Missing Jsdoc for this method
</blockquote></details>

Done, hope I made it right

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/prestashop/prestashop/8938)*

**Comment by Quetzacoalt91** [comment]
*[admin-dev/themes/default/js/bundle/module/module.js, line 662 at r4](https://beta.reviewable.io/reviews/prestashop/prestashop/8938#-L9zQzLZ489o5OKEUWF-:-L9zhL0z3tqXh5RjjALW:b-2nqumr) ([raw file](https://github.com/prestashop/prestashop/blob/9aada29086b7351d14ea861cbee5bbbb0fc59f83/admin-dev/themes/default/js/bundle/module/module.js#L662)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

What if the request fails?
</blockquote></details>

Well, I don't expect the interface to do change anything in case of error.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/prestashop/prestashop/8938)*

**Comment by Quetzacoalt91** [comment]
*[admin-dev/themes/default/js/bundle/module/module.js, line 665 at r4](https://beta.reviewable.io/reviews/prestashop/prestashop/8938#-L9zR8wf33NEUISl8iU5:-L9zhZwWEFU-ZqjwOlTt:b6194su) ([raw file](https://github.com/prestashop/prestashop/blob/9aada29086b7351d14ea861cbee5bbbb0fc59f83/admin-dev/themes/default/js/bundle/module/module.js#L665)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

```js
'a.tab[href="'+badge.target+'"]'
```
Isn't there another way, like using ids?
</blockquote></details>

Unfortunately no, level 3 tabs are not displayed with unique attributes:



---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/prestashop/prestashop/8938)*

**Comment by Quetzacoalt91** [comment]
*[install-dev/data/xml/tab.xml, line 183 at r4](https://beta.reviewable.io/reviews/prestashop/prestashop/8938#-L9zRU1b9jZp7F4Cbq0i:-L9zhuV_2gchgetwPi1V:b-hdhp0p) ([raw file](https://github.com/prestashop/prestashop/blob/9aada29086b7351d14ea861cbee5bbbb0fc59f83/install-dev/data/xml/tab.xml#L183)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

There's an indentation issue compared to the old lines
</blockquote></details>

And this is expected. If you look at the whole file, we played with tabulation to show the different levels of sub-menus.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/prestashop/prestashop/8938)*

**Comment by Quetzacoalt91** [comment]
I took advantage of this PR still opened to add the ID attr in the header tabs. This will be easier for me to add the notification count on the proper tab, and later for the E2E tests for its selectors

@eternoendless, that was one of your comments :)

**Comment by eternoendless** [comment]
Reviewed 5 of 5 files at r5.
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[admin-dev/themes/default/js/bundle/module/module.js, line 658 at r4](https://beta.reviewable.io/reviews/prestashop/prestashop/8938#-L9zRzoS20TmoLEheFpZ:-LAnOTTNEVSUQvp8xQra:blgr70) ([raw file](https://github.com/prestashop/prestashop/blob/9aada29086b7351d14ea861cbee5bbbb0fc59f83/admin-dev/themes/default/js/bundle/module/module.js#L658)):*
<details><summary><i>Previously, Quetzacoalt91 (Thomas N) wrote…</i></summary><blockquote>

Done, hope I made it right
</blockquote></details>

Thanks! Keep in mind that you don't need to add `@return` if it's a void function

---

*[admin-dev/themes/default/js/bundle/module/module.js, line 662 at r4](https://beta.reviewable.io/reviews/prestashop/prestashop/8938#-L9zQzLZ489o5OKEUWF-:-LAnOfNxDaXxfrUtg-Tb:beivzsk) ([raw file](https://github.com/prestashop/prestashop/blob/9aada29086b7351d14ea861cbee5bbbb0fc59f83/admin-dev/themes/default/js/bundle/module/module.js#L662)):*
<details><summary><i>Previously, Quetzacoalt91 (Thomas N) wrote…</i></summary><blockquote>

Well, I don't expect the interface to do change anything in case of error.
</blockquote></details>

Shouldn't it display an error?

---

*[install-dev/data/xml/tab.xml, line 183 at r4](https://beta.reviewable.io/reviews/prestashop/prestashop/8938#-L9zRU1b9jZp7F4Cbq0i:-LAnP5pr13MiT_S0msHX:b-w9sxr8) ([raw file](https://github.com/prestashop/prestashop/blob/9aada29086b7351d14ea861cbee5bbbb0fc59f83/install-dev/data/xml/tab.xml#L183)):*
<details><summary><i>Previously, Quetzacoalt91 (Thomas N) wrote…</i></summary><blockquote>

And this is expected. If you look at the whole file, we played with tabulation to show the different levels of sub-menus.
</blockquote></details>

Ok then

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/prestashop/prestashop/8938)*

**Comment by eternoendless** [comment]
Thanks for the ids :thumbsup: 

---

Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/prestashop/prestashop/8938#-:-LAnPOqj0u5aJtT9Q8Xh:b6de188)*

**Comment by Quetzacoalt91** [comment]
* Add a generic error message in console
* Update selectors in e2e tests

**Comment by eternoendless** [comment]
:lgtm:

---

Reviewed 4 of 4 files at r6, 2 of 2 files at r7, 1 of 1 files at r8.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8938#-:-LBupX9S-1KZlESUvWhy:bnfp4nl)*

**Review by eternoendless — DISMISSED** [review]

**Review by eternoendless — APPROVED** [review]

**Comment by eternoendless** [comment]
Thank you @Quetzacoalt91

**Pull request merged**